src/improvements/tasks/MultisigTask.sol

```diff
@@ -615,52 +615,47 @@ abstract contract MultisigTask is Test, Script, StateOverrideManager {
-                    continue;
-                }
-                console.log("Nested multisig: %s", getAddressLabel(startingOwners[i]));
-                console.logBytes(getEncodedTransactionData(startingOwners[i], callData));
-            }
-        }
+        console.log("Child multisig: %s", getAddressLabel(childMultisig));
+        // Logs required for using eip712sign binary to sign the data to sign with Ledger
+        console.log("vvvvvvvv");
+        console.logBytes(getEncodedTransactionData(childMultisig, callData));
+        console.log("^^^^^^^^\n");
     }
 
-    /// @notice print the hash to approve by EOA for nested multisig
+    /// @notice Print the hash to approve by EOA for nested multisig.
     function printChildHash(Action[] memory actions, address childMultisig) public view {
-        bytes memory callData = generateApproveMulticallData(actions);
-
-        // this branch means the function `signFromChildMultisig` is being called
-        if (childMultisig != address(0)) {
-            console.log("Nested multisig: %s", getAddressLabel(childMultisig));
-            console.logBytes32(keccak256(getEncodedTransactionData(childMultisig, callData)));
-        } else {
-            // this branch means function `signFromChildMultisig` is not being called
-            // and this is not a nested safe
-            for (uint256 i; i < startingOwners.length; i++) {
-                // do not get data to sign if owner is an EOA (not a multisig)
-                if (startingOwners[i].code.length == 0) {
-                    continue;
-                }
+        require(
+            childMultisig != address(0), "MultisigTask: Child multisig cannot be zero address when printing child hash."
+        );
 
-                bytes32 hash = keccak256(getEncodedTransactionData(startingOwners[i], callData));
-                console.log("Nested multisig: %s", getAddressLabel(startingOwners[i]));
-                console.logBytes32(hash);
-            }
+        bytes memory callData = generateApproveMulticallData(actions);
+        console.log("Child multisig: %s", getAddressLabel(childMultisig));
+
+        bytes memory encodedTxData = getEncodedTransactionData(childMultisig, callData);
+        require(encodedTxData.length == 66, "MultisigTask: encodedTxData length is not 66 bytes.");
+
+        bytes32 safeTxHash;
+        assembly {
+            // 66 bytes = (bytes1(0x19), bytes1(0x01), bytes32(domainSeparator()), bytes32(safeTxHash))
+            // Retrieve the last 32 bytes of encodedTxData (safeTxHash).
+            // Memory layout of encodedTxData:
+            // - The first 32 bytes store the length (66 bytes in this case).
+            // - The actual data starts at encodedTxData + 32.
+            // - The last 32 bytes of the data (safeTxHash) start at:
+            //   encodedTxData + 32 + (66 - 32) = encodedTxData + 66.
+            safeTxHash := mload(add(encodedTxData, mload(encodedTxData)))
```

Comment:
This is what eip712sign is doing: https://github.com/base/eip712sign/blob/main/main.go#L114